airmozilla/popcorn/renderer.py

```diff
@@ -0,0 +1,92 @@
+import tempfile
+import os
+import timezone
+
+from airmozilla.base.utils import build_absolute_url, prepare_vidly_video_url
+from airmozilla.popcorn.models import PopcornEdit
+from airmozilla.main.models import Event, VidlySubmission
+from airmozilla.manage import vidly
+from airmozilla.uploads.models import Upload
+
+from django.db import transaction
+from django.conf import settings
+
+import boto
+from boto.s3.key import Key
+from funfactory.urlresolvers import reverse
+from popcoder.popcoder import process_json
+
+
+@transaction.atomic
+def render_edit(id):
+    edit = PopcornEdit.objects.get(id=id)
+    event = edit.event
+
+    filename = '%s.webm' % edit.id
+    filepath = os.path.join(tempfile.gettempdir(), filename)
+    filesize = os.stat(filepath)[os.stat.st_size]
+    if not (os.path.isfile(filepath) and filesize > 0):
+        edit.status = PopcornEdit.STATUS_PROCESSING
+        edit.save()
+        process_json(edit.data, out=filepath)
+
+    # Uploads file to s3
+    connection = boto.connect_s3(
+        settings.AWS_ACCESS_KEY_ID,
+        settings.AWES_SECRET_ACCESS_KEY,
+    )
+    bucket = connection.get_bucket('vidly_uploads')
+    video_key = Key(bucket)
+
+    video_key.key = filename
```

Comment:
This needs to change. Either here or on line 25. 
We need something random. 
Otherwise you'd be able to guess that the URL is `https://s3.amazon.com/airmo/1.webm` or `https://s3.amazon.com/airmo/<SOME INTEGER>.webm` and you'd be able to just `wget` integers until you luckily manage to download a private video that you shouldn't be able to `wget`.